web/src/test/java/org/springframework/security/web/savedrequest/SimpleSavedRequestTests.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.savedrequest;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import javax.servlet.http.Cookie;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SimpleSavedRequestTests {
+
+	private SimpleSavedRequest savedRequest;
+
+	@Before
+	public void setUp() {
+		SavedRequest savedRequestMock = mock(SavedRequest.class);
```

Comment:
Instead of a mock, if you constructed a `SimpleSavedRequest` with a redirect URL and then called the setters, you'd get a similar outcome while at the same time testing more of the class.

Reply:
Improved as suggested